packages/react/src/internal/useControllableState.ts

```diff
@@ -36,7 +36,7 @@ export function useControllableState({
   value,
 }) {
   const [state, internalSetState] = useState(value ?? defaultValue);
-  const controlled = useRef(null);
+  const controlled = useRef<boolean | null>(null);
```

Comment:
could we solve for this using JSDocs instead? we generally don't want to convert internal files to typescript 
From our docs: https://github.com/carbon-design-system/carbon/blob/main/docs/guides/adding-component-types.md#strategy
<img width="1045" alt="image" src="https://user-images.githubusercontent.com/40550942/234036335-9b7b73a6-a4d1-4ffe-99b2-595ca6caf7a2.png">